From b3d4e7c5a2f8196f49da1c03e7f5a28b0d6f42e8 Mon Sep 17 00:00:00 2001
From: Gabriel Przybysz Gonçalves Júnior <gabriel.przybsyzgoncalvesjunior@windriver.com>
Date: Wed, 8 Oct 2025 09:45:02 -0300
Subject: [PATCH] csi-powerstore: add snapshotclass in templates

This patch consists of adding a snapshot class in the helm chart
templates directory.
It's needed because this helm chart needs a snapshot class that should
be created manually.

Signed-off-by: Gabriel Przybysz Gonçalves Júnior <gabriel.przybsyzgoncalvesjunior@windriver.com>
---
 charts/csi-powerstore/templates/snapshotclass.yaml | 6 ++++++
 charts/csi-powerstore/values.yaml                  | 1 +
 2 files changed, 7 insertions(+)
 create mode 100644 charts/csi-powerstore/templates/snapshotclass.yaml

diff --git a/charts/csi-powerstore/templates/snapshotclass.yaml b/charts/csi-powerstore/templates/snapshotclass.yaml
new file mode 100644
index 0000000..e3ab58c
--- /dev/null
+++ b/charts/csi-powerstore/templates/snapshotclass.yaml
@@ -0,0 +1,6 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: {{ .Release.Name }}-snapshot
+driver: {{ .Values.driverName | default ( print "csi-powerstore.dellemc.com" ) }}
+deletionPolicy: {{ .Values.controller.snapshot.deletionPolicy }}
diff --git a/charts/csi-powerstore/values.yaml b/charts/csi-powerstore/values.yaml
index a636e8f..694e0f0 100644
--- a/charts/csi-powerstore/values.yaml
+++ b/charts/csi-powerstore/values.yaml
@@ -147,6 +147,7 @@ controller:
     # Default value: None
     enabled: true
     image: registry.k8s.io/sig-storage/csi-snapshotter:v8.3.0
+    deletionPolicy: Delete

     # snapNamePrefix: Prefix to apply to the names of a created snapshots
     # Allowed values: string
--
2.34.1